app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [
+        Person(person_dict["name"], person_dict["age"])
+        for person_dict in people
+    ]
+
+    for person_dict in people:
+        person = Person.people[person_dict["name"]]
+        if person_dict.get("wife"):
+            person.wife = Person.people[person_dict["wife"]]
```

Comment:
You could use walrus operator for optimization as well
```suggestion
        if wife_name := person_dict.get("wife"):
            person.wife = Person.people[wife_name]
```